io.mjml.mjml-app.yaml

```diff
@@ -0,0 +1,74 @@
+app-id: io.mjml.mjml-app
+base: io.atom.electron.BaseApp
+base-version: '18.08'
+runtime: org.freedesktop.Platform
+runtime-version: '18.08'
+sdk: org.freedesktop.Sdk
+command: mjml-app
+finish-args:
+  - --socket=x11
+  - --share=ipc
+  - --share=network
+  - --device=dri
+  - --filesystem=home
+modules:
+  - name: mjml-app
+    buildsystem: simple
+    build-commands:
+      - install -Dt /app/bin apply_extra mjml-app
+      - install -Dm644 -t /app/share/metainfo io.mjml.mjml-app.appdata.xml
+      - install -Dm644 -t /app/share/applications io.mjml.mjml-app.desktop
+      - install -Dm644 -T 16x16.png /app/share/icons/hicolor/16x16/apps/io.mjml.mjml-app.png
+      - install -Dm644 -T 32x32.png /app/share/icons/hicolor/32x32/apps/io.mjml.mjml-app.png
+      - install -Dm644 -T 48x48.png /app/share/icons/hicolor/48x48/apps/io.mjml.mjml-app.png
+      - install -Dm644 -T 128x128.png /app/share/icons/hicolor/128x128/apps/io.mjml.mjml-app.png
+      - install -Dm644 -T 256x256.png /app/share/icons/hicolor/256x256/apps/io.mjml.mjml-app.png
+      - install -Dm644 -T 512x512.png /app/share/icons/hicolor/512x512/apps/io.mjml.mjml-app.png
+      - install -Dm644 -T 1024x1024.png /app/share/icons/hicolor/1024x1024/apps/io.mjml.mjml-app.png
+    sources:
+      - type: script
+        dest-filename: apply_extra
+        commands:
+          - mkdir _tmp
+          - mv *.AppImage _tmp
+          - cd _tmp
+          - chmod +x *.AppImage
+          - ./*.AppImage --appimage-extract
+          - mv squashfs-root ../mjml-app
+          - cd ..
+          - rm -rf _tmp
+          - chmod -R 755 mjml-app
+      - type: script
+        dest-filename: mjml-app
+        commands:
+          - exec /app/extra/mjml-app/app/mjml-app "$@"
+      - type: file
+        path: io.mjml.mjml-app.desktop
+      - type: file
+        path: io.mjml.mjml-app.appdata.xml
+      - type: extra-data
+        filename: mjml-app-2.10.0-linux-x86_64.AppImage
+        url: https://github.com/mjmlio/mjml-app/releases/download/v2.10.0/mjml-app-2.10.0-linux-x86_64.AppImage
+        sha256: 576a27b1307ad58629a6f9d815f8639cd7505ed8e6a72b6c5f10a827af6ec40d
+        size: 58398922
+      - type: file
+        url: https://github.com/mjmlio/mjml-app/raw/master/build/icons/16x16.png
```

Comment:
would a stable commit id or a tag rather than "master" be better here to make builds in the future a little more robust?  Or is breakage intended when these files change upstream?

Reply:
It's just a png file.

Reply:
You are probably right, those should be pinpointed to the tag of the release I'm using, just to be futureproof. I will have to see If downloading these icons seperately will be necessary at all when I rework it to not use the appimage.